pkg/bucketrequest/bucketrequest.go

```diff
@@ -80,27 +80,29 @@ func (b *bucketRequestListener) provisionBucketRequestOperation(ctx context.Cont
 	// Most code here is identical to that found in controller.go of kube's  controller...
 	bucketClassName := b.GetBucketClass(bucketRequest)
 
-	//  A previous doProvisionBucketRequest may just have finished while we were waiting for
-	//  the locks. Check that bucket (with deterministic name) hasn't been provisioned
-	//  yet.
-	bucket := b.FindBucket(ctx, bucketRequest)
-	if bucket != nil {
-		// bucket has been already provisioned, nothing to do.
-		glog.Info("Bucket already exists", bucket.Name)
-		return util.ErrBucketAlreadyExists
-	}
-
 	bucketClass, err := b.bucketClient.ObjectstorageV1alpha1().BucketClasses().Get(ctx, bucketClassName, metav1.GetOptions{})
 	if bucketClass == nil {
 		// bucketclass does not exist in order to create a bucket
 		return util.ErrInvalidBucketClass
 	}
 
-	glog.Infof("Creating Bucket for BucketRequest %v", bucketRequest.Name)
+	name := bucketRequest.Spec.BucketPrefix
+	if name != "" {
+		name = name + "-"
+	}
+	name = name + string(bucketRequest.GetUID())
+	name = "br-" +  name
 
+	bucket, err := b.bucketClient.ObjectstorageV1alpha1().Buckets().Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
If the bucket is found, then err == nil and we'll continue to process.  We would want to abort wouldn't we?

Reply:
I agree, FindBucket above returns util.ErrBucketAlreadyExists which makes the controller to discard this request.